lib/integrations/externalauditstorage/configurator.go

```diff
@@ -109,6 +111,13 @@ func (o *Options) setDefaults(ctx context.Context, region string) error {
 			config.WithRegion(region),
 			config.WithUseFIPSEndpoint(useFips),
 			config.WithRetryMaxAttempts(10),
```

Comment:
hmm... before we try all this request/response body logging... do you mind just deleting this `WithRetryMaxAttempts` line? I had copied it from some WIP code tobiasz had, and I honestly don't know the reason for it. I bet it's actually what's causing us to go over the 30s context deadline and just get useless errors. Hopefully we get good error messages without it 🤞 I'm pretty sure there's no good reason at all for us to have it

looks like the default max attempts is 3 with a max backoff of 4 seconds (i may be off by a factor of 2 but it should be under 30s for sure)

Reply:
You're correct. When I remove the max attempts lines, log message switches over to: 
```
Failed to retrieve new credentials: failed to retrieve credentials, operation error STS: AssumeRoleWithWebIdentity, exceeded maximum number of attempts, 3, https response error StatusCode: 400, RequestID: GUID, InvalidIdentityToken: Incorrect token audience
```